### PR TITLE
[MSHARED-1158] make DependencyCollectorBuilder nore configurable

### PR DESCRIPTION
**Pull request by olamy**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    private ArtifactFilter filter;
+
+    private Map<String, Object> configProperties = new HashMap<>();
+
+    private DependencySelector dependencySelector = new AndDependencySelector(
+            new DirectScopeDependencySelector( JavaScopes.TEST ),
+            new DirectScopeDependencySelector( JavaScopes.PROVIDED ),
+            new OptionalDependencySelector(),
+            new ExclusionDependencySelector() );
```
Comment (olamy) [inline src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java:93]: well TBH this empty ExclusionDependencySelector doesn't have any effect. we could simply remove it and the code not will not pass tru to do nothing :)

File: src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+ *             new SimpleOptionalitySelector(),
+ *             new JavaScopeDeriver() );
+ * </pre>
+ * </div>
+ * <div>
+ * configProperties have 2 default values
+ * <pre>
+ *   ConflictResolver.CONFIG_PROP_VERBOSE, true
+ *   DependencyManagerUtils.CONFIG_PROP_VERBOSE, true
+ * </pre>
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java:75]: I assume that we mention properties for resolver configuration, maybe will be good idea put a link to
https://maven.apache.org/resolver/configuration.html

**Review by slawekjaranowski — COMMENTED** [review]
I think that we build `Repository System Session` from scratch here ... can we use prepared session by Maven ... in `org.apache.maven.internal.aether.DefaultRepositorySystemSessionFactory#newRepositorySession`
It is to investigate ...

File: src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java (hunk 2 above)
Comment (olamy) [reply src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java:75]: Good idea!

**Comment by olamy** [comment]
> I think that we build `Repository System Session` from scratch here ... can we use prepared session by Maven ... in `org.apache.maven.internal.aether.DefaultRepositorySystemSessionFactory#newRepositorySession`
> 
> It is to investigate ... 

Yup but should be part of another change?

File: src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java (hunk 2 above)
Comment (olamy) [reply src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java:75]: done in 8f4c9f63bf985d858e5e11456f612e9479d561cc

**Pull request merged**